### PR TITLE
get_managed_identity_source() for Azure Identity

### LINKED ISSUE #697: [Feature Request] Need an API for Managed Identity to detect the current environment
### MSAL client type

Confidential

### Problem Statement

Please refer to the "MSAL implementation" part of [this internal PR](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?version=GBdev&path=/ManagedIdentity/FacilitateImdsProbing.md&_a=preview&anchor=msal-implementations).

### Proposed solution

_No response_

### PR DESCRIPTION
**Pull request by rayluo**
Reference: [Internal specs](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?version=GBdev&path=/ManagedIdentity/FacilitateImdsProbing.md&_a=preview&anchor=msal-implementations)

### CODE REVIEW

File: msal/managed_identity.py [hunk 1]
```diff
@@ -288,6 +289,29 @@ def _scope_to_resource(scope):  # This is an experimental reasonable-effort appr
     return scope  # There is no much else we can do here
 
 
+APP_SERVICE = object()
+AZURE_ARC = object()
+CLOUD_SHELL = object()
+MACHINE_LEARNING = object()
+SERVICE_FABRIC = object()
+DEFAULT_TO_VM = object()  # Unknown environment; default to VM; you may want to probe
+def get_managed_identity_source():
```
Comment (neha-bhargava) [inline msal/managed_identity.py:310]: Not very familiar with the MSAL python code but shouldn't there be unit tests for this flow?

**Review by neha-bhargava — APPROVED** [review]
Approved with comments

File: msal/managed_identity.py (hunk 1 above)
Comment (rayluo) [reply msal/managed_identity.py:310]: Did not aim for 100% [test coverage](https://stackoverflow.com/questions/90002/what-is-a-reasonable-code-coverage-for-unit-tests-and-why). But, yeah, new test cases is added now.

**Comment by rayluo** [comment]
Shipped in [MSAL Python 1.29.0](https://github.com/AzureAD/microsoft-authentication-library-for-python/releases/tag/1.29.0)

**Pull request merged**